test/crdsvalidation/kongkeyset/testcases/common.go

```diff
@@ -0,0 +1,34 @@
+package testcases
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	configurationv1alpha1 "github.com/kong/kubernetes-configuration/api/configuration/v1alpha1"
+)
+
+// testCase is a test case related to KongKeySet validation.
+type testCase struct {
+	Name                       string
+	KongKeySet                 configurationv1alpha1.KongKeySet
+	KongKeySetStatus           *configurationv1alpha1.KongKeySetStatus
+	Update                     func(*configurationv1alpha1.KongKeySet)
+	ExpectedErrorMessage       *string
+	ExpectedUpdateErrorMessage *string
+}
+
+type testCasesGroup struct {
+	Name      string
+	TestCases []testCase
+}
+
+// TestCases is a collection of all test cases groups related to KongKeySet validation.
+var TestCases = []testCasesGroup{}
+
+func init() {
+	TestCases = append(TestCases, keySetAPISpec, cpRef)
+}
```

Comment:
Since you're specifying explicitly all tests here where the definition of `TestCases` you can add all of them as an item to a list without `init`. On the other hand `init` can be put where the test case is defined to have its registration in the same place as definition of a case. WDYT?

Reply:
I believe this can be a nice improvement but I'd stay consistent with other tests suites for now.